subprojects/core/src/main/java/org/gradle/internal/work/DefaultAsyncWorkTracker.java

```diff
@@ -92,6 +92,22 @@ public void run() {
         }
     }
 
+    @Override
+    public boolean hasUncompletedWork(BuildOperationRef operation) {
+        lock.lock();
+        try {
+            List<AsyncWorkCompletion> workItems = ImmutableList.copyOf(items.get(operation));
+            return CollectionUtils.any(workItems, new Spec<AsyncWorkCompletion>() {
```

Comment:
I personally prefer a simple for-loop below Java 8. It just creates less noise.

Reply:
Sure, why not.  Done.